### PR TITLE
[Dashboard] Better support per worker gpu usage from the cluster view. 

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

**I manually tested**. Please let me know if you have good ideas to unit test this.

Also cc @alanwguo I think we may want to unit test these components... Is it possible for you or Chao to take it? 

After we fixed the issue from the infra where per worker GPU usage is not recorded, it seems like per worker GPU stat is already working.

However, I found several problems from existing code, and this PR fixes issues. It includes;
- Right now, instead of gpu.index, it uses a random arbitrary index. I fixed this issue so that the GPU index matches the output of gpustat (i.e., nvidia-smi)
- Per worker GPU usage shows the logical GPU allocation. I fixed it so that it will now display the physical usage (which is consistent with the node row). However, GPUstat cannot figure out the "per proc" GPU usage, meaning the GPU usage per proc will be identical to the GPU usage per node for a given index (device). 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
LGTM. Thanks!

**Review by alanwguo — COMMENTED** [review]
Re: testing.

We've been adding tests as we go along to all the new code. I agree that we should add some tests to some existing pages. I think the jobs page and the clusters page will need a test.

File: dashboard/client/src/pages/node/GPUColumn.tsx [hunk 1]
```diff
@@ -74,32 +48,30 @@ export const NodeGPUView = ({ node }: { node: NodeDetail }) => {
...
+      return <NodeGPUEntry key={gpu.uuid} gpu={gpu} slot={gpu.index} />;
+    })
+    .filter((entry) => entry !== undefined);
+
+  return workerGPUEntries.length === 0 ? (
+    <Typography color="textSecondary" component="span" variant="inherit">
+      N/A
+    </Typography>
+  ) : (
+    <div style={{ minWidth: GPU_COL_WIDTH }}>{workerGPUEntries}</div>
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/client/src/pages/node/GPUColumn.tsx:75]: will do!

[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/client/src/pages/node/GPUColumn.tsx:75]: ADdressed

**Review by alanwguo — APPROVED** [review]

**Pull request merged**